prototypes/output_slice/output_slice_parallel.cc

```diff
@@ -0,0 +1,1584 @@
+                                           error_per_cell,
+                                           VectorTools::L2_norm);
+}
+
+template <int dim, int fe_degree>
+void
+MatrixBasedPoissonProblem<dim, fe_degree>::output_results(
+  const unsigned int cycle)
+{
+  if (triangulation.n_global_active_cells() > 1e6)
+    return;
+
+  TimerOutput::Scope t(computing_timer, "output complete");
+
+  Timer output_timer;
+  output_timer.start();
+
+  solution.update_ghost_values();
+
+  DataOut<dim> data_out;
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(solution, "solution");
+
+  Vector<float> subdomain(triangulation.n_active_cells());
+  for (unsigned int i = 0; i < subdomain.size(); ++i)
+    {
+      subdomain(i) = triangulation.locally_owned_subdomain();
+    }
+  data_out.add_data_vector(subdomain, "subdomain");
+
+  data_out.build_patches(mapping, fe.degree, DataOut<dim>::curved_inner_cells);
+
+  DataOutBase::VtkFlags flags;
+  flags.compression_level = DataOutBase::CompressionLevel::best_speed;
+  data_out.set_flags(flags);
+  data_out.write_vtu_with_pvtu_record(parameters.output_path,
+                                      parameters.output_name +
+                                        std::to_string(dim) + "d",
+                                      cycle,
+                                      MPI_COMM_WORLD,
+                                      3);
+}
+
+template <int dim, int fe_degree>
+void
+MatrixBasedPoissonProblem<dim, fe_degree>::output_slice_results(
+  const unsigned int cycle)
+{
+  if (triangulation.n_global_active_cells() > 1e6)
+    return;
+
+  TimerOutput::Scope t(computing_timer, "output slice");
+
+  Timer output_timer;
+  output_timer.start();
+
+  solution.update_ghost_values();
+
+  DataOutResample<dim, dim - 1, dim> data_out(patch_triangulation,
+                                              patch_mapping);
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(dof_handler, solution, "solution");
+
+  Vector<float> subdomain(triangulation.n_active_cells());
+  for (unsigned int i = 0; i < subdomain.size(); ++i)
+    {
+      subdomain(i) = triangulation.locally_owned_subdomain();
+    }
+  data_out.add_data_vector(subdomain, "subdomain");
+
+  pcout << "building patches for slice output" << std::endl;
+
+  data_out.build_patches(mapping,
+                         fe.degree,
+                         DataOut<dim - 1, dim>::curved_inner_cells);
+  pcout << "slice output" << std::endl;
+  DataOutBase::VtkFlags flags;
+  flags.compression_level = DataOutBase::CompressionLevel::best_speed;
+  data_out.set_flags(flags);
+  data_out.write_vtu_with_pvtu_record(parameters.output_path,
+                                      parameters.output_name +
+                                        std::to_string(dim) + "d" + "_slice",
+                                      cycle,
+                                      MPI_COMM_WORLD,
```

Comment:
In the end this is the only part you want to test :). Where the solution comes from does not matter :)